app/models/miq_worker/systemd_common.rb

```diff
@@ -11,6 +11,30 @@ def supports_systemd?
         false
       end
 
+      def systemd_manager
+        @systemd_manager ||= begin
+          require "dbus/systemd"
+          DBus::Systemd::Manager.new
+        end
+      end
+
+      def services
+        systemd_manager.units.select { |unit| unit[:name].start_with?(service_name) }
+      end
+
+      def active_services
+        services.select { |service| service[:active_state] == "active" }.map { |service| service[:name] }
+      end
+
+      def failed_services
+        services.select { |service| service[:active_state] == "failed" }.map { |service| service[:name] }
```

Comment:
This is cool.  I've started doing something similar with kubernetes and categorizing failed ones as failed and then removing the deployments and allowing new ones to start makes me wonder if we could get into an infinite loop if each new deployment will continue to fail because we're detecting one should be started but for whatever reason, the runner itself can't start properly.  I have to think about that some more.